var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -605,6 +606,37 @@ def cmake_args(self):
         define = self.define
         define_from_variant = self.define_from_variant
 
+        if self.spec.satisfies("@master:"):
+            ver_in_src = os.path.exists(
+                os.path.join(self.stage.source_path, "packages", "kokkos", "CMakeLists.txt")
+            )
+            if ver_in_src:
+                with open(
+                    os.path.join(self.stage.source_path, "packages", "kokkos", "CMakeLists.txt")
+                ) as f:
+                    all_txt = f.read()
+                    r = dict(
+                        re.findall(
+                            r".*set\s?\(\s?Kokkos_VERSION_(MAJOR|MINOR|PATCH)\s?(\d+)", all_txt
+                        )
+                    )
+                    KokkosVersionInTrilinosSource = Version(
```

Comment:
objects should be `snake_case` in python. You can also de-indent everything after `all_txt` since it's not necessary to keep the file open. And is the file check guard on 613 necessary? Maybe replace the earlier condition with `"@master: +kokkos"` to be more specific. If kokkos is required, that source path better be there...

Reply:
Sure, I was trying to catch the case where sometimes users with incorrect proxy settings may download .html or .xml files when requests for .tar.gz resources fail (so the file would be missing). However the checksum would catch this case too, so as you suggested, I went ahead and dropped that check.